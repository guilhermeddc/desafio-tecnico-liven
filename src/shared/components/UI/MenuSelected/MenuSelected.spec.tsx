import {render, screen} from '@testing-library/react';

import {MenuSelected} from '.';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn(),
}));

describe('MenuSelected', () => {
  it('renders correctly', () => {
    const {container} = render(
      <MenuSelected active={true} onClick={() => null}>
        Home
      </MenuSelected>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    render(
      <MenuSelected active={true} onClick={() => null}>
        Home
      </MenuSelected>,
    );

    expect(screen.getByText('Home')).toBeInTheDocument();
  });
});
