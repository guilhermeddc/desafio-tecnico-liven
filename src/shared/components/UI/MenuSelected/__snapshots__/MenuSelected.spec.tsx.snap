// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuSelected renders correctly 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-l8f8kw-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <div
        class="css-4wcyto-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-1id1zev-MuiSvgIcon-root"
          data-testid="ArrowDropUpRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.71 12.29 11.3 9.7c.39-.39 1.02-.39 1.41 0l2.59 2.59c.63.63.18 1.71-.71 1.71H9.41c-.89 0-1.33-1.08-.7-1.71z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-jwshyv-MuiSvgIcon-root"
          data-testid="ArrowDropDownRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m8.71 11.71 2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
          />
        </svg>
      </div>
    </span>
    Home
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
